src/android/kiwixreader.cpp

```diff
@@ -234,21 +234,24 @@ JNIEXPORT jbyteArray JNICALL Java_org_kiwix_kiwixlib_JNIKiwixReader_getContent(
   jbyteArray data = env->NewByteArray(0);
 
   /* Retrieve the content */
-  std::string cUrl = jni2c(url, env);
+  std::string cUrl = getStringObjValue(url, env);
   unsigned int cSize = 0;
 
   try {
     auto entry = READER->getEntryFromEncodedPath(cUrl);
+    bool isRedirect = entry.isRedirect();
     entry = entry.getFinalEntry();
     cSize = entry.getSize();
     setIntObjValue(cSize, sizeObj, env);
-
-    data = env->NewByteArray(cSize);
-    env->SetByteArrayRegion(
-        data, 0, cSize, reinterpret_cast<const jbyte*>(entry.getBlob().data()));
-
     setStringObjValue(entry.getMimetype(), mimeTypeObj, env);
     setStringObjValue(entry.getTitle(), titleObj, env);
+    if (isRedirect) {
+      setStringObjValue(entry.getPath(), url, env);
```

Comment:
Shouldn't this be along the lines of `getFinalPath()`? It doesn't seem to change the URL for me at the moment

Reply:
Ok it seem that entry.getPath() is correct. Still not getting the right output in android though.

Reply:
My bad it seems intellij wasn't updating the variable.